c/C90/C90.g4

```diff
@@ -0,0 +1,672 @@
+    notice, this list of conditions and the following disclaimer.
+ 2. Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimer in the
+    documentation and/or other materials provided with the distribution.
+ 3. The name of the author may not be used to endorse or promote products
+    derived from this software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+grammar C90;
+
+compilationUnit
+    :   externalDeclaration* EOF
+    ;
+
+externalDeclaration
+    :   functionDefinition
+    |   functionDefinitionKandR
+    |   functionDeclaration
+    |   'extern'? variableDeclaration
+    |   typeDeclaration ';'
+    |   typeDefinition
+    |   typeWillBeDeclared
+    ;
+
+storageFuncSpecifier
+    :   'static'
+    |   '__inline__'
+    |   'extern'
+    ;
+
+storageVarSpecifier
+    :   'register'
+    ;
+
+
+typeWillBeDeclared
+    : ('struct'|'union') Identifier ';'
+    ;
+
+functionSpecifier
+    :  '__inline'
+    ;
+
+//if typeSpecifier not spedified : default return int
+functionDeclaration
+    :   '__extension__'? gccDeclaratorExtension1* storageFuncSpecifier* functionSpecifier? gccDeclaratorExtension1* typeSpecifier? gccDeclaratorExtension1* visualExtension? Identifier function gccDeclaratorExtension2* ';'
+    ;
+
+functionDefinition
+    :   '__extension__'? gccDeclaratorExtension1* storageFuncSpecifier* functionSpecifier? gccDeclaratorExtension1* typeSpecifier? gccDeclaratorExtension1* Identifier '(' parameterList ')' compoundStatement
+    ;
+
+functionDefinitionKandR
+    :   typeSpecifier Identifier '(' varList? ')' (typeSpecifier Identifier ';')* compoundStatement
+    ;
+
+varList
+    : variableName (',' variableName)*
+    ;
+
+typeQualifier
+    :   'const'
+    |   '__const'
+    |   '__restrict'
+    |   '__restrict__'
+    |   'volatile'
+    ;
+
+typeDeclaration
+    : structDeclaration
+    | enumDeclaration
+    ;
+
+typeSpecifier
+    :   (storageVarSpecifier | typeQualifier)* typeName typeModifier* typeQualifier*
+    ;
+
+type
+    : typeName
+    | typeQualifier typeName
+    | typeDeclaration
+    ;
+
+typeName
+    : 'int'
+    | ('unsigned'|'signed')? 'long' 'int'?
```

Comment:
I suggest extracting `'unsigned'|'signed'` to the separated rule since it's used in many places:

```g4
unsignedOrSigned
    : 'unsigned'
    | 'signed'
    ;
```

```suggestion
    | unsignedOrSigned? 'long' 'int'?
```